service/raft/group0_state_id_handler.cc

```diff
@@ -14,8 +15,19 @@ namespace service {
 
 static logging::logger slogger("group0_tombstone_gc_handler");
 
-group0_state_id_handler::group0_state_id_handler(gms::gossiper& gossiper)
-    : _gossiper(gossiper) {
+group0_state_id_handler::group0_state_id_handler(replica::database& local_db, gms::gossiper& gossiper)
+    : _local_db(local_db)
+    , _gossiper(gossiper) {
+    _gossiper.register_(shared_from_this());
+}
+
+group0_state_id_handler::~group0_state_id_handler() {
+    SCYLLA_ASSERT(_stopped);
```

Comment:
Pleas don't use assert, we are trying to purge them from the codebase.
Use `on_internal_fatal_error()` instead.

And please also consider: is the fact that `_stopped = false` here such a fatal error that it is worth bringing the node down? Or is it enough to just log an error?

Reply:
but... we are bringing the node down. (Since we are in this destructor)

Reply:
done